cppapi/server/device_4.cpp

```diff
@@ -40,7 +40,28 @@
 #include <ac_config.h>
 #endif
 
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#endif
+
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wsometimes-uninitialized"
+#pragma clang diagnostic ignored "-Wunused-variable"
+#endif
+
```

Comment:
There are many files including tango.h header file.
Only cppapi/server/device_4.cpp and cppapi/server/device_5.cpp got a patch like this.
cppapi/server/device.cpp includes tango.h too for instance.
@t-b , Do you know why only cppapi/server/device_4.cpp and cppapi/server/device_5.cpp needed this patch?
This is not clear to me.

Reply:
@bourtemb I presume because only these two cpp files use the code which triggers the warnings.

Reply:
@bourtemb You can't review and approve because you opened the PR. Please merge if it is good.

Reply:
Github also gives me the possibility to review...

This PR is fixing all the warnings when using the latest gcc but it is also introducing the following warning for Debian < 10:

```
/home/tango/src/cpp_test_suite/cxxtest/include/../../new_tests/cxx_old_poll.cpp:906:32: warning: unknown option after '#pragma GCC diagnostic' kind [-Wpragmas]

 #pragma GCC diagnostic ignored "-Wcatch-value"

                                ^~~~~~~~~~~~~~~
```

This is not a critical issue of course. There are some other warnings reported in these old Debian releases.

When testing with CLion, I noticed there are currently some inconsistencies in the values of the CMAKE_BUILD_TYPE CMake variable.
There are several locations in the cppTango CMakefiles where CMAKE_BUILD_TYPE value is compared to "DEBUG" or "RELEASE" values instead of comparing it to "Debug" or "Release".
CMAKE_BUILD_TYPE is even set to "RELEASE" by default.
I think this should be changed to use "Release" and "Debug" values, which are actually some of the possible values documented in the official CMake documentation and used by CLion:
https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html
This issue can be observed in the Travis builds where CMAKE_BUILD_TYPE is set to "Release".
As you can see, -Og option is used when it should not be in this use case:
https://travis-ci.org/tango-controls/cppTango/jobs/612484597

Reply:
https://github.com/bourtemb/cppTango/commit/923ecfeeb6cea6d581faba598dbd9423d63b64f8 commit in my fork should solve these CMAKE_BUILD_TYPE inconsistencies issues.
I can create a dedicated PR for that.

Reply:
@bourtemb Yes old gcc compilers (< 8.0) don't know about that "Wcatch-value". I could add some version magic and I'm also wondering if it is worth the trouble.